src/cswinrt/strings/additions/Windows.Storage.Streams/IMarshal.cs

```diff
@@ -210,43 +219,81 @@ private static int Do_Abi_DisconnectObject_5(IntPtr thisPtr, uint dwReserved)
+        public IMarshal(IObjectReference obj)
+        {
+            _obj = obj.As<global::WinRT.Interop.IUnknownVftbl>(global::WinRT.Interop.IID.IID_IMarshal);
+        }
+#else
         public IMarshal(IObjectReference obj) : this(obj.As<Vftbl>()) { }
         internal IMarshal(ObjectReference<Vftbl> obj)
         {
             _obj = obj;
         }
+#endif
 
         public unsafe void GetUnmarshalClass(Guid* riid, IntPtr pv, global::Com.MSHCTX dwDestContext, IntPtr pvDestContext, global::Com.MSHLFLAGS mshlFlags, Guid* pCid)
         {
+#if NET
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, Guid*, IntPtr, global::Com.MSHCTX, IntPtr, global::Com.MSHLFLAGS, Guid*, int>)(*(void***)thisPtr)[3])(thisPtr, riid, pv, dwDestContext, pvDestContext, mshlFlags, pCid));
+#else
             Marshal.ThrowExceptionForHR(_obj.Vftbl.GetUnmarshalClass_0(ThisPtr, riid, pv, dwDestContext, pvDestContext, mshlFlags, pCid));
+#endif
         }
 
         public unsafe void GetMarshalSizeMax(Guid* riid, IntPtr pv, global::Com.MSHCTX dwDestContext, IntPtr pvDestContext, global::Com.MSHLFLAGS mshlflags, uint* pSize)
         {
+#if NET
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, Guid*, IntPtr, global::Com.MSHCTX, IntPtr, global::Com.MSHLFLAGS, uint*, int>)(*(void***)thisPtr)[4])(thisPtr, riid, pv, dwDestContext, pvDestContext, mshlflags, pSize));
+#else
             Marshal.ThrowExceptionForHR(_obj.Vftbl.GetMarshalSizeMax_1(ThisPtr, riid, pv, dwDestContext, pvDestContext, mshlflags, pSize));
+#endif
         }
 
         public unsafe void MarshalInterface(IntPtr pStm, Guid* riid, IntPtr pv, global::Com.MSHCTX dwDestContext, IntPtr pvDestContext, global::Com.MSHLFLAGS mshlflags)
         {
+#if NET
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, IntPtr, Guid*, IntPtr, global::Com.MSHCTX, IntPtr, global::Com.MSHLFLAGS, int>)(*(void***)thisPtr)[5])(thisPtr, pStm, riid, pv, dwDestContext, pvDestContext, mshlflags));
+#else
             Marshal.ThrowExceptionForHR(_obj.Vftbl.MarshalInterface_2(ThisPtr, pStm, riid, pv, dwDestContext, pvDestContext, mshlflags));
+#endif
         }
 
         public unsafe void UnmarshalInterface(IntPtr pStm, Guid* riid, IntPtr* ppv)
         {
+#if NET
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, IntPtr, Guid*, IntPtr*, int>)(*(void***)thisPtr)[6])(thisPtr, pStm, riid, ppv));
+#else
             Marshal.ThrowExceptionForHR(_obj.Vftbl.UnmarshalInterface_3(ThisPtr, pStm, riid, ppv));
+#endif
         }
 
         public unsafe void ReleaseMarshalData(IntPtr pStm)
         {
+#if NET
+            IntPtr thisPtr = ThisPtr;
+            Marshal.ThrowExceptionForHR(((delegate* unmanaged[Stdcall]<IntPtr, IntPtr, int>)(*(void***)thisPtr)[6])(thisPtr, pStm));

```

Comment:
indexing seems wrong starting from here.  There is 2 6.

Reply:
Good catch! I assume this also means we are missing some tests?

Reply:
I believe we have tests for the overall scenario, but maybe not this specific function.  I am not sure which specific scenario that this IMarshal function gets called in and if it even does for how we use it.  Will need to take a look.

Reply:
Fixed in https://github.com/microsoft/CsWinRT/pull/1597/commits/4ffd01739cf83b5c5020826a4e74f7e15905b759.